sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachinesImpl.java

```diff
@@ -274,6 +282,72 @@ public Accepted<Void> beginDeleteByResourceGroup(String resourceGroupName, Strin
                 Context.NONE);
     }
 
+    @Override
+    public PagedIterable<VirtualMachine> listByVirtualMachineScaleSetId(String vmssId) {
+        return new PagedIterable<>(this.listByVirtualMachineScaleSetIdAsync(vmssId));
+    }
+
+    @Override
+    @SuppressWarnings({"unchecked", "removal"})
+    public PagedFlux<VirtualMachine> listByVirtualMachineScaleSetIdAsync(String vmssId) {
+        if (CoreUtils.isNullOrEmpty(vmssId)) {
+            return new PagedFlux<>(() -> Mono.error(
+                new IllegalArgumentException("Parameter 'vmssId' is required and cannot be null.")));
+        }
+        // Hack in nextLink encoding by using reflection.
+        // Replace below hack with "listAsync()" once backend fix "nextLink" encoding issue:
+        // https://github.com/Azure/azure-rest-api-specs/issues/25640
+        Method listSinglePageAsync;
+        try {
+            listSinglePageAsync = inner().getClass().getDeclaredMethod("listByResourceGroupSinglePageAsync", String.class, String.class, ExpandTypeForListVMs.class);
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+        Method listNextSinglePageAsync;
+        try {
+            listNextSinglePageAsync = inner().getClass().getDeclaredMethod("listNextSinglePageAsync", String.class, Context.class);
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+        java.security.AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
```

Comment:
why do we need `AccessController.doPrivileged`?

`AccessController` is to be removed, if this gets in, it had to be temporary solution.

Reply:
Yeah, it's required by CI (DP_DO_INSIDE_DO_PRIVILEGED).

Reply:
OK, we still want service to fix this issue. We would avoid this as much as possible in long run.

Reply:
Sure. Is it OK we use reflection here and switch to `Copy` afterwards(if service doesn't fix it then)? 

`ExpandableStringEnum` uses `ReflectionUtils.getLookupToUse`, which use `AccessController` as well:
https://github.com/Azure/azure-sdk-for-java/blob/b3c847da07ca9d1e3c1869d6d259b3fa291adc43/sdk/core/azure-core/src/main/java/com/azure/core/util/ExpandableStringEnum.java#L88
https://github.com/Azure/azure-sdk-for-java/blob/b3c847da07ca9d1e3c1869d6d259b3fa291adc43/sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionUtils.java#L171
We could switch implementation before core does.

Reply:
Ok, if core uses it, I guess we can (at least temporary).